ocaml/xapi/xapi_message.ml

```diff
@@ -817,10 +817,10 @@ let handler (req : Http.Request.t) fd _ =
           (ExnHelper.string_of_exn e)
   )
 
-(* Export messages and send to another host/pool over http. *)
-let send_messages ~__context ~cls ~obj_uuid ~session_id ~remote_address =
-  let msgs = get ~__context ~cls ~obj_uuid ~since:(Date.of_float 0.0) in
-  let body = export_xml msgs in
+(* Export and send host messages to another host/pool over http. *)
+let send_messages ~__context ~cls ~obj_uuid ~session_id ~remote_address
+    ~messages =
+  let body = export_xml messages in
   let query =
     [
       ("session_id", Ref.string_of session_id); ("cls", "VM"); ("uuid", obj_uuid)
```

Comment:
There was a latent bug here, the class is ahrdcoded, it should be possible to convert it to a string
```suggestion
      ("session_id", Ref.string_of session_id); ("cls", to_string cls); ("uuid", obj_uuid)
```